user/models.py

```diff
@@ -1,5 +1,42 @@
+from django.contrib.auth.base_user import BaseUserManager
 from django.contrib.auth.models import AbstractUser
+from django.db import models
+from django.utils.translation import gettext as _
+
+
+class UserManager(BaseUserManager):
+    use_in_migrations = True
+
+    def _create_user(self, email, password, **extra_fields):
+        if not email:
+            raise ValueError("The given email and password")
+        email = self.normalize_email(email)
+        user = self.model(email=email, **extra_fields)
+        user.set_password(password)
+        user.save(using=self._db)
+        return user
+
+    def create_user(self, email, password=None, **exra_fields):
+        exra_fields.setdefault("is_staff", False)
+        exra_fields.setdefault("is_superuser", False)
+        return self._create_user(email, password, **exra_fields)
+
+    def create_superuser(self, email, password, **extra_fields):
+        extra_fields.setdefault("is_staff", True)
+        extra_fields.setdefault("is_superuser", True)
+
+        if extra_fields.get("is_staff") is not True:
+            raise ValueError("Superuser must have is_stall=True")
```

Comment:
There is a typo in your error message. You wrote 'is_stall' instead of 'is_staff'. It's important to provide accurate error messages to help with debugging if an error occurs.